tests/topotests/bgp_duplicate_nexthop/test_bgp_duplicate_nexthop.py

```diff
@@ -0,0 +1,456 @@
+    switch.add_link(tgen.gears["r6"])
+
+    switch = tgen.add_switch("s14")
+    switch.add_link(tgen.gears["r1"])
+
+
+def setup_module(mod):
+    "Sets up the pytest environment"
+    tgen = Topogen(build_topo, mod.__name__)
+    tgen.start_topology()
+
+    router_list = tgen.routers()
+
+    for rname, router in router_list.items():
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_ISIS, os.path.join(CWD, "{}/isisd.conf".format(rname))
+        )
+        if rname in ("r1", "r3", "r5", "r6"):
+            router.load_config(
+                TopoRouter.RD_BGP, os.path.join(CWD, "{}/bgpd.conf".format(rname))
+            )
+
+    # Initialize all routers.
+    tgen.start_router()
+
+
+def teardown_module(_mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+
+    tgen.stop_topology()
+
+
+def check_ipv4_prefix_with_multiple_nexthops(prefix, multipath=True):
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    logger.info(
+        f"Check that {prefix} unicast entry is installed with paths for r5 and r6"
+    )
+
+    r5_nh = [
+        {
+            "ip": "192.0.2.5",
+            "active": True,
+            "recursive": True,
+        },
+        {
+            "ip": "172.31.0.3",
+            "interfaceName": "r1-eth1",
+            "active": True,
+            "labels": [
+                16055,
+            ],
+        },
+        {
+            "ip": "172.31.2.4",
+            "interfaceName": "r1-eth2",
+            "active": True,
+            "labels": [
+                16055,
+            ],
+        },
+    ]
+
+    r6_nh = [
+        {
+            "ip": "192.0.2.6",
+            "active": True,
+            "recursive": True,
+        },
+        {
+            "ip": "172.31.0.3",
+            "interfaceName": "r1-eth1",
+            "active": True,
+            "labels": [
+                16006,
+            ],
+        },
+        {
+            "ip": "172.31.2.4",
+            "interfaceName": "r1-eth2",
+            "active": True,
+            "labels": [
+                16006,
+            ],
+        },
+    ]
+
+    expected = {
+        prefix: [
+            {
+                "prefix": prefix,
+                "protocol": "bgp",
+                "metric": 0,
+                "table": 254,
+                "nexthops": [],
+            }
+        ]
+    }
+    for nh in r5_nh:
+        expected[prefix][0]["nexthops"].append(nh)
+    if multipath:
+        for nh in r6_nh:
+            expected[prefix][0]["nexthops"].append(nh)
+
+    test_func = functools.partial(
+        ip_check_path_selection, tgen.gears["r1"], prefix, expected
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, f"Failed to check that {prefix} uses the IGP label 16055"
```

Comment:
But it can be the label 16006 also, correct?

Reply:
right. fixed comment